PWGJE/EMCALJetTasks/AliEmcalJetTask.cxx

```diff
@@ -1057,6 +1066,50 @@ void AliEmcalJetTask::LoadTrackEfficiencyFunction(const std::string & path, cons
   delete file;
 }
 
+/**
+ * Set the pt-dependent tracking efficiency depending on periodfrom a YAML file
+ * @param path Path to the file containing the TF1
+ * @param name Name of the TF1
+ */
+
+void AliEmcalJetTask::SetTrackingEfficiencyFromYAML(std::string period , std::string path ) {
```

Comment:
Could you elaborate on this? I think I understand what you mean, that one would have to run over separate multi-period runs individually. But its not clear to me how one could do this as you suggest, would it be possible to get the name of the period on-the-fly such that each event selects the correct tracking uncertainty?

Reply:
Thanks a lot for the explanation, that makes more sense now., and I think that is a good idea. Does this function get the data period from the run number, and not the MC period?
I can test anyway and let you know how I get on.

Reply:
I think these values are all defined when the data are produced, so it's not from the run number. It looks like `ALIEN_JDL_LPMPRODUCTIONTAG` (or the equivalent in metadatasets) is the MC or data period, but there's also a `ALIEN_JDL_LPMPRODUCTIONTAG`. 

Briefly looking back at the code that I linked, the function that I linked isn't so careful about metadatasets, so this might need a bit more logic. I explicitly noted in that code `  // We assume that all children in the metadataset should have the same collision system and production type.` So even this approach would need some care. I don't have the capacity to think through this in detail right now, but you can look at an example like [here](http://alitrain.cern.ch/train-workdir/PWGJE/Jets_EMC_pp_MC/2845_20220607-1515/__TRAIN__/env.sh) to see what is available. Maybe we can think through together what might work?

Reply:
Sorry I was offline at the end of last week and am just getting round to looking at this. I'm wondering whether this method ( doing something like gSystem->Getenv("ALIEN_JDL_LPMPRODUCTIONTAG"); ) would only work if running on the trains and this variable ALIEN_JDL.. is set, right? At least when I run locally it doesn't work. Would this be an issue?

One alternative would be to take the run info from the input event - I found an example in the conversion code of AOD->AO2D:

https://github.com/alisw/AliPhysics/blob/15aeb623f5c3d3bfce51d59ceef723372911bd91/RUN3/AliAnalysisTaskAO2Dconverter.cxx#L463-L470

Here the anchor period would be accessible (as well as other information, in case we wanted to map the MC period instead, though I think the anchor period would be best, as is currently done) . As I understand it, the data would not be loaded at the AddTask stage, so this step would have to be done 'later' on - perhaps SetArtificialTrackingEfficiencyFromYAML() could be executed instead when the first event is run, e.g. in AliEmcalJetTask::Run() , conditionally checking whether fTrackEfficiencyHistogram exists, or something similar. This method of getting the anchor period works with a local check anyway.

Let me know your thoughts anyway

Reply:
> 

just responding to your suggestion on the centrality:
Centrality bins can’t be obtained from the task and are hard-coded in AliAnalysisTaskEmcal::RetrieveEventObjects.  So I added a check on the number of centrality bins and confirm that this is equal to either 4 or 5, or is running over pp (where this doesn’t matter)